src/app/screens/LNURLRedeem/index.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  CaretLeftIcon,
+  CrossIcon,
+  QrCodeIcon,
+} from "@bitcoin-design/bitcoin-icons-react/filled";
+import Button from "@components/Button";
+import Container from "@components/Container";
+import Header from "@components/Header";
+import IconButton from "@components/IconButton";
+import QrcodeScanner from "@components/QrcodeScanner";
+import TextField from "@components/form/TextField";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import lnurlLib from "~/common/lib/lnurl";
+import { isLNURLDetailsError } from "~/common/utils/typeHelpers";
+
+function LNURLRedeem() {
+  const { t } = useTranslation("translation", { keyPrefix: "lnurlredeem" });
+
+  const [invoice, setInvoice] = useState("");
+  const navigate = useNavigate();
+  const [qrIsOpen, setQrIsOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    try {
+      setLoading(true);
+
+      const lnurl = lnurlLib.findLnurl(invoice);
+
+      if (lnurl) {
+        const lnurlDetails = await lnurlLib.getDetails(lnurl);
+
+        if (isLNURLDetailsError(lnurlDetails)) {
+          toast.error(lnurlDetails.reason);
+          return;
+        }
+
+        if (lnurlDetails.tag === "withdrawRequest") {
+          navigate("/lnurlWithdraw", {
+            state: {
+              args: {
+                lnurlDetails,
+              },
+            },
+          });
+        } else {
+          toast.error("Not a lnurl-withdraw request");
+          return;
+        }
+      } else {
+        toast.error("Not a valid LNURL");
+        return;
+      }
+    } catch (e) {
+      if (e instanceof Error) {
+        toast.error(e.message);
+      }
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  function extractInvoiceFrom(data: string) {
```

Comment:
this will be used if we scan qr codes which contains  url where lnurl-w code is given in the tag named ```lightning```

eg. https://legend.lnbits.com/?lightning=LNURL1DP68GURN8GHJ7MR9VAJKUEPWD3HXY6T5WVHXXMMD9AMKJARGV3EXZAE0V9CXJTMKXYHKCMN4WFKZ7KZDD4E455ECGAJ8Q5698P55W6M8DGENSATK9AT52K26GAU5CJJ32ET5K6RWTP38WATSDDK5CL2QW0E